Sources/Logging/Logging.swift

```diff
@@ -654,22 +694,34 @@ let systemStdout = Darwin.stdout
 #elseif os(Windows)
 let systemStderr = CRT.stderr
 let systemStdout = CRT.stdout
-#else
+#elseif canImport(Glibc)
 let systemStderr = Glibc.stderr!
 let systemStdout = Glibc.stdout!
+#elseif canImport(WASILibc)
+// no stdio on WASI yet
+#else
+#error("Unsupported runtime")
 #endif
 
 /// `StreamLogHandler` is a simple implementation of `LogHandler` for directing
 /// `Logger` output to either `stderr` or `stdout` via the factory methods.
 public struct StreamLogHandler: LogHandler {
     /// Factory that makes a `StreamLogHandler` to directs its output to `stdout`
     public static func standardOutput(label: String) -> StreamLogHandler {
-        return StreamLogHandler(label: label, stream: StdioOutputStream.stdout)
+        #if canImport(WASILibc)
+            return StreamLogHandler(label: label, stream: PrintOutputStream())
+        #else
+            return StreamLogHandler(label: label, stream: StdioOutputStream.stdout)
+        #endif
     }
 
     /// Factory that makes a `StreamLogHandler` to directs its output to `stderr`
     public static func standardError(label: String) -> StreamLogHandler {
-        return StreamLogHandler(label: label, stream: StdioOutputStream.stderr)
+        #if canImport(WASILibc)
+            return standardOutput(label: label)
+        #else
+            return StreamLogHandler(label: label, stream: StdioOutputStream.stderr)
+        #endif
```

Comment:
I don't think these two changes will be needed if suggestions above are applied.